src/tensor.cpp

```diff
@@ -430,6 +430,18 @@ std::size_t TensorDescriptor::GetNumBytes() const
 
 bool TensorDescriptor::IsPacked() const { return this->packed; }
 
+bool TensorDescriptor::IsSameLength(const TensorDescriptor& otherDesc) const
```

Comment:
You can simply use this.GetLengths() != otherDesc.GetLengths() to check. Because GetLengths() basically return std::vector and in C++ you can compare 2 vectors directly with == and != operator

Reply:
updated